packages/unmock-hash/src/hash/v0/index.ts

```diff
@@ -2,31 +2,27 @@ import objectHash from "object-hash";
 import querystring from "querystring";
 import { IJSONValue } from "../../jsonType";
 
-export interface IHashableV0 {
-  body: string | {};
-  headers: { [key: string]: string };
-  hostname: string;
-  method: string;
-  path: string;
-  story: string[];
-  user_id: string;
-  signature?: string;
+interface IStringMap {
+  [key: string]: string;
 }
 
-export interface IHashableWithIgnoreV0 {
-  body?: string | {};
-  headers?: { [key: string]: string };
-  hostname?: string;
-  method?: string;
-  path?: string;
-  story?: string[];
-  user_id?: string;
+export interface IHashableV0 {
```

Comment:
IMO, we don't want any fields in `IHashable` to be undefined except fields, like signature, that can be legitimately undefined.  For example, if I understand correctly, this would allow someone to pass an empty object à la `computeHash({})` and it to return something.  The ideal behavior would be ts not even compiling that code.

As a side note, IIRC, `foo: string | undefined` is the same as `foo?: string`.

Reply:
I understand the logic behind that - although it would still produce a valid hash.
Hence the compromise; `foo: string | undefined` is not the same as `foo?: string` - the latter is an optional field, while the former is a mandatory field whose value might be undefined.
At worse, it would allow someone to pass:
```typescript
computeHash({
  body: undefined,
  headers: undefined,
  mostname: undefined,
  method: undefined,
  path: undefined,
  story: undefined,
  user_id: undefined,
});
```

The bonus for it is allowing one less interface which IMO clutters the namespace.

Reply:
`Partial<T>` to the rescue!